feat(quickstarts): Laravel and Rails. (#20264)
supabase/supabase: apps/docs/pages/guides/getting-started/quickstarts/laravel.mdx

import Layout from '~/layouts/DefaultGuideLayout'
import StepHikeCompact from '~/components/StepHikeCompact'

export const meta = {
  title: 'Use Supabase with Laravel',
  subtitle:
    'Learn how to create a PHP Laravel project, connect it to your Supabase Postgres database, and configure user authentication.',
  breadcrumb: 'Framework Quickstarts',
}

<StepHikeCompact>

  <StepHikeCompact.Step step={1}>
    <StepHikeCompact.Details title="Create a Laravel Project">

    Make sure your PHP and Composer versions are up to date, then use `composer create-project` to scaffold a new Laravel project.

>     For more details you can head over to the [Laravel docs](https://laravel.com/docs/10.x/installation#creating-a-laravel-project).

Reviewer: ```suggestion
    See the [Laravel docs](https://laravel.com/docs/10.x/installation#creating-a-laravel-project) for more details.
```

    </StepHikeCompact.Details>

    <StepHikeCompact.Code>

     ```bash Terminal
     composer create-project laravel/laravel example-app
     ```

    </StepHikeCompact.Code>

  </StepHikeCompact.Step>

  <StepHikeCompact.Step step={2}>
    <StepHikeCompact.Details title="Install the Authentication template">

>     Install [Laravel Breeze](https://laravel.com/docs/10.x/starter-kits#laravel-breeze) which is minimal, simple implementation of all of Laravel's [authentication features](https://laravel.com/docs/10.x/authentication).

Reviewer: ```suggestion
    Install [Laravel Breeze](https://laravel.com/docs/10.x/starter-kits#laravel-breeze), a simple implementation of all of Laravel's [authentication features](https://laravel.com/docs/10.x/authentication).
```

    </StepHikeCompact.Details>

    <StepHikeCompact.Code>

    ```bash Terminal
    composer require laravel/breeze --dev
    php artisan breeze:install
    ```

    </StepHikeCompact.Code>

  </StepHikeCompact.Step>

  <StepHikeCompact.Step step={3}>
    <StepHikeCompact.Details title="Set up the Postgres connection details">

>     Head over to [database.new](https://database.new) and create a new Supabase project. Make sure to secure save your database password.

Reviewer: ```suggestion
    Go to [database.new](https://database.new) and create a new Supabase project. Save your database password securely.
```

    When your project is up and running, navigate to the [database settings](https://supabase.com/dashboard/project/_/settings/database) to find the URI connection string.

>     Laravel ships with a Postgres adapter out of the box, you can simply configure it via the environment variables. You can find the database URL in your [Supabase Dashboard](/dashboard/project/_/settings/database).

Reviewer: ```suggestion
    Laravel ships with a Postgres adapter out of the box. Configure it with your Supabase environment variables.
```

    </StepHikeCompact.Details>

    <StepHikeCompact.Code>

    ```bash .env
    DB_CONNECTION=pgsql
    DATABASE_URL=postgresql://postgres:[YOUR-PASSWORD]@db.[YOUR-PROJECT-REF].supabase.co:5432/postgres
    ```

    </StepHikeCompact.Code>

  </StepHikeCompact.Step>

  <StepHikeCompact.Step step={4}>
    <StepHikeCompact.Details title="Run the database migrations">

    Laravel ships with database migration files that set up the required tables for Laravel Authentication and User Management.

    Note: Laravel does not use Supabase Auth but rather implements its own authentication system!

    </StepHikeCompact.Details>

    <StepHikeCompact.Code>

    ```bash Terminal
    php artisan migrate
    ```

    </StepHikeCompact.Code>

  </StepHikeCompact.Step>

  <StepHikeCompact.Step step={5}>
    <StepHikeCompact.Details title="Start the app">

>     Run the development server, go to http://127.0.0.1:8000 in a browser and see your application running. You can also nvaigate to http://127.0.0.1:8000/register and http://127.0.0.1:8000/login to register and/or login users.

Reviewer: ```suggestion
    Run the development server. Go to http://127.0.0.1:8000 in a browser to see your application. You can also nvaigate to http://127.0.0.1:8000/register and http://127.0.0.1:8000/login to register and log in users.
```
